pyk/src/pyk/kast/manip.py

```diff
@@ -128,7 +129,17 @@ def _ml_constraint_to_bool(_kast: KInner) -> KInner:
                 if second == FALSE:
                     return notBool(first)
                 if type(first) in [KVariable, KToken]:
-                    return KApply('_==K_', _kast.args)
+                    assert hasattr(first, 'sort')
+                    if first.sort == KSort('Int'):
+                        return KApply('_==Int_', _kast.args)
+                    else:
+                        return KApply('_==K_', _kast.args)
+                if type(second) in [KVariable, KToken]:
+                    assert hasattr(second, 'sort')
+                    if second.sort == KSort('Int'):
+                        return KApply('_==Int_', _kast.args)
+                    else:
+                        return KApply('_==K_', _kast.args)
```

Comment:
In this way, we make sure that we do not use `==K` if we know we can use `==Int`.